src/lib/telldus-accessory.js

```diff
@@ -164,12 +178,39 @@ class TelldusDimmer extends TelldusSwitch {
    */
   getDimLevel(callback, context) {
     this.log('getDimLevel called')
+    
     TDtool.device(this.id).then(device => {
-      if (device.dimlevel)
-        return callback(null, bitsToPercentage(parseInt(device.dimlevel)))
-      callback(null, device.lastsentcommand === 'ON' ? 100 : 0)
+      if (device.dimlevel){
+        this.dimlevel = bitsToPercentage(parseInt(device.dimlevel))
+        return callback(null, this.dimlevel)
+      }
+      callback(null, this.dimlevel)
     })
+  }
 
+  /**
+   * Set the state of this Accessory. This accepts a value that is either
+   * true or false, for turning the dimmer off or on to previous set dimmer level.
+   *
+   * @param  {*}                  value          The value to set,
+   *                                             corresponding to the passed
+   *                                             Characteristic
+   * @param  {Function}           callback       To be invoked when result is
+   *                                             obtained.
+   * @param  {object}             context
+   */
+  setState(value, callback, context) {
+    this.log(`Recieved set state request for dimmer: [${value ? 'on' : 'off'}]`)
+    if (value && this.dimlevel > 0){
+      TDtool.dim(percentageToBits(this.dimlevel), this.id).then(out => {
```

Comment:
I believe a return statement is needed here, or line `213` will always be executed.